presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -802,6 +802,36 @@ public void finishInsert(ConnectorSession session, ConnectorInsertTableHandle in
             for (CompletableFuture<?> fileRenameFuture : fileRenameFutures) {
                 MoreFutures.getFutureValue(fileRenameFuture, PrestoException.class);
             }
+
+            // clean remaining partition directories
+            Optional<Path> writePathRoot = locationService.writePath(handle.getLocationHandle(), Optional.empty());
+            if (writePathRoot.isPresent() && !writePathRoot.get().equals(locationService.targetPathRoot(handle.getLocationHandle()))) {
+                FileSystem fileSystem;
+                try {
+                    fileSystem = hdfsEnvironment.getFileSystem(session.getUser(), writePathRoot.get());
+                }
+                catch (IOException e) {
+                    throw new PrestoException(HIVE_FILESYSTEM_ERROR, e);
+                }
+                for (PartitionUpdate partitionUpdate : partitionUpdates) {
+                    verify(!partitionUpdate.getTargetPath().equals(partitionUpdate.getWritePath()));
+                    Path writePath = new Path(partitionUpdate.getWritePath());
```

Comment:
What about this commit: https://github.com/prestodb/presto/pull/3851/commits/79d2bafb6d1f8792ca5cb47c88509dcda75d9dfb

Havining this, you woundn't need to wrap wrap `writePath` as `Path` each time. IMO  this commit makes `PartitionUpdate` be coherent with `LocatioService` as in that case both would return  `Path` for `wrtiePath` and `targetPath`. Moreover  it is more intuitive to return `Path` for paths.